scipy/stats/_bws_test.py

```diff
@@ -167,13 +194,13 @@ def bws_test(x, y, *, alternative="two-sided", method=None):
     difference in performance between the two groups.
     '''
 
-    x, y, alternative, method = _bws_input_validation(x, y, alternative,
-                                                      method)
-    bws_statistic = partial(_bws_statistic, alternative=alternative)
+    x, y, alternative, axis, method, xp = _bws_input_validation(x, y, alternative,
+                                                                axis, method)
+    bws_statistic = partial(_bws_statistic, alternative=alternative, xp=xp)
 
     permutation_alternative = 'less' if alternative == 'less' else 'greater'
     res = stats.permutation_test((x, y), bws_statistic,
```

Comment:
`bws_test` does whatever `permutation_test` does with NaNs and empty input. I noticed that `permutation_test` didn't have a test with NaN input and the behavior was not ideal[^1], so I fixed it and added a test in this PR.

[^1]: The p-value was `0` for an exact test or `1 / (n_resamples + 1)` for a randomized permutation test because zero elements of the null distribution are more extreme than NaN... but the same could be said of any stats function. We shouldn't return a numerical p-value with a NaN statistic.